packages/loot-core/src/shared/arithmetic.ts

```diff
@@ -43,11 +47,10 @@ function parsePrimary(state) {
   // TODO: The regex should probably respect the number format better,
   // and we should do more strict parsing
   let numberStr = '';
+  let thousandsSep = getNumberFormat().separator === ',' ? '.' : ',';
   while (char(state) && char(state).match(/[0-9,.]/)) {
-    let thousandsSep = getNumberFormat().separator === ',' ? '.' : ',';
-
     // Don't include the thousands separator
-    if (char(state) === thousandsSep) {
+    if (remainingLength(state) >= 4 && char(state) === thousandsSep) {
```

Comment:
Can you add a comment here clarifying that this is to make it format agnostic?

Reply:
New revision should be clearer.